block-layer/codecs/dag-pb.md

```diff
@@ -36,6 +36,21 @@ message PBNode {
 The objects link names are specified in the 'Name' field of the PBLink object.
 All link names in an object must either be omitted or unique within the object.
 
+This layout can be expressed with [IPLD Schemas](../../schemas/README.md) as:
+
+```ipldsch
+type PBLink struct {
+  hash optional Bytes
```

Comment:
Should this be `optional` when we describe it with an IPLD Schema?  I think not.

Name apparently *is* optional, the prose above suggests.  Tsize... I think also tends towards being considered technically optional, though I sort of don't want to touch that discussion again.  But the hash itself?  Pretty sure if that's not present, it will quickly result in nope.

I suspect `hash` is only optional in the protobuf because of the weird "everything should be optional" meme in protobuf lands, and I don't think we should carry that into IPLD Schema descriptions because we don't have any of the same problems nor ergonomics around optionality that protobuf does.

Reply:
From a schema standpoint, I wouldn't treat it as optional. Size/name are definitely optional (both technically and logically from a schema standpoint).